llvm-external-projects/iree-dialects/lib/Dialect/LinalgTransform/IR/StructuredTransformOpsExt.cpp

```diff
@@ -387,7 +387,8 @@ static Operation *findSingleDefiningOp(Operation *replacedOp,
       .Case<scf::ForOp>([&](scf::ForOp) -> Operation * {
         return findSingleForOpDefiningAll(range);
       })
-      .Default([](Operation *) -> Operation * { return nullptr; });
+      .Default(
+          [&](Operation *) -> Operation * { return range[0].getDefiningOp(); });
```

Comment:
This would change the semantics of the function: it currently returns an defining operation that defines _all_ values in the range if it exists; with the change, it will return the defining operation of _first_ value and ignore the rest that may be defined by other operations. IIRC, it is important that _all_ values are replaced by the same operation because of the internal workings of the rewriter/listener. This could work if the code checked that there is only one value in the range.